### PR TITLE
Update Turbine computer to use TGUI

### PR DESCRIPTION
**Pull request by psykzz**
The old interface could use some improvement.

:cl: psykzz
add: Added TGUI for Turbine computer
/:cl:

image

### CODE REVIEW

**Comment by GunHog** [comment]
Neat :+1:

File: code/modules/power/turbine.dm [hunk 1]
```diff
@@ -331,51 +331,38 @@
...
 
-	var/datum/browser/popup = new(user, "turbinecomputer", name)
-	popup.set_content(dat)
-	popup.open()
-	return
+	data["working"] = (src.compressor.starter && compressor && compressor.turbine && !compressor.stat && !compressor.turbine.stat)
+	data["connected"] = (compressor && compressor.turbine) ? TRUE : FALSE
+	data["compressor_broke"] = (!compressor || compressor.stat) ? TRUE : FALSE
+	data["turbine_broke"] = (!compressor || compressor.turbine.stat) ? TRUE : FALSE
+	data["online"] = src.compressor.starter
```
Comment (Cyberboss) [inline code/modules/power/turbine.dm]: remove src

File: code/modules/power/turbine.dm [hunk 2]
```diff
@@ -331,51 +331,38 @@
...
-		if(!compressor)
-			dat += "<A href='?src=[REF(src)];search=1'>Search for compressor</A>"
+/obj/machinery/computer/turbine_computer/ui_data(mob/user)
+	var/list/data = list()
 
-	var/datum/browser/popup = new(user, "turbinecomputer", name)
-	popup.set_content(dat)
-	popup.open()
-	return
+	data["working"] = (src.compressor.starter && compressor && compressor.turbine && !compressor.stat && !compressor.turbine.stat)
```
Comment (Cyberboss) [inline code/modules/power/turbine.dm]: remove src

File: code/modules/power/turbine.dm [hunk 3]
```diff
@@ -331,51 +331,38 @@
...
+	data["compressor_broke"] = (!compressor || compressor.stat) ? TRUE : FALSE
+	data["turbine_broke"] = (!compressor || compressor.turbine.stat) ? TRUE : FALSE
+	data["online"] = src.compressor.starter
 
-/obj/machinery/computer/turbine_computer/Topic(href, href_list)
-	if(..())
-		return
+	data["power"] = DisplayPower(src.compressor.turbine.lastgen)
+	data["rpm"] = src.compressor.rpm
+	data["temp"] = src.compressor.gas_contained.temperature
```
Comment (Cyberboss) [inline code/modules/power/turbine.dm]: remove src

**Review by Cyberboss — CHANGES_REQUESTED** [review]
:+1: You're doing gods work, son

**Comment by psykzz** [comment]
@Cyberboss removed

**Review by Cyberboss — DISMISSED** [review]

File: tgui/src/interfaces/turbine_computer.ract [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+		{{#if data.working}}
+			<span class='{{powerState(data.online)}}'>{{data.online && !(data.compressor_broke || data.turbine_broke) ? "Online" : "Offline"}}</span>
+		{{else}}
+			<span class='bad'>Broken</span>
+		{{/if}}
+		<div style='float:right'>
+			<ui-button icon='power-off' action='power' state='{{data.working}}' style='{{data.online ? "selected" : ""}}'>On</ui-button>
+			<ui-button icon='close' action='power' state='{{data.working}}' style='{{data.online ? "" : "selected"}}'>Off</ui-button>
+		</div>
+		{{#if data.compressor_broke}}
```
Comment (JJRcop) [inline tgui/src/interfaces/turbine_computer.ract]: Use `||` here instead of two ifs.

Comment (psykzz) [reply tgui/src/interfaces/turbine_computer.ract]: 2nd line should actually say Turbine instead, changing that.

**Review by Cyberboss — APPROVED** [review]

File: tgui/src/interfaces/turbine_computer.ract (hunk 4 above)
Comment (JJRcop) [reply tgui/src/interfaces/turbine_computer.ract]: resolved

**Pull request merged**